src/services/rfqm_service.ts

```diff
@@ -726,39 +754,28 @@ export class RfqmService {
         workerAddress: string,
         callData: string,
     ): Promise<SubmissionsMap> {
-        // make sure this order hasn't been used by another worker (or this worker before)
-        if ((await this._dbUtils.findRfqmTransactionSubmissionsByOrderHashAsync(orderHash)).length !== 0) {
-            throw new Error('found pre-existing transaction submissions');
-        }
+        let submissionsMap: SubmissionsMap;
 
-        let submissionsMap: SubmissionsMap = {};
+        const previousSubmissions = await this._dbUtils.findRfqmTransactionSubmissionsByOrderHashAsync(orderHash);
 
-        let gasPrice: BigNumber;
-        let nonce: number;
-        let gasEstimate: number;
+        // if there are previous tx submissions (repair mode), pick up monitoring and re-submission
+        // else initaliaze the submission context and send the first tx
+        const submissionContext =
+            previousSubmissions.length > 0
+                ? await this._recoverSubmissionsContextAsync(workerAddress, orderHash, callData, previousSubmissions)
+                : await this._initalizeSubmissionsContextAsync(workerAddress, orderHash, callData);
```

Comment:
beautiful

(of course, I'm extremely biased)

Reply:
I strongly prefer this as well